aplib.net-demo/Assets/Scripts/Points/KeyPointsAdder.cs

```diff
@@ -0,0 +1,17 @@
+using Assets.Scripts.Items;
+using UnityEngine;
+
+public class KeyPointsAdder : PointsAdderComponent
+{
+    private void Awake()
+    {
+        PickupableKey item = GetComponent<PickupableKey>();
+        item.KeyPickedUp += SendPoints;
+    }
```

Comment:
You should be subscribing to and unsubscribing from events in OnEnable() and OnDisable.